File: src/main/java/main/commands/slashcommands/commands/dev/PostCommandInfoCommand.java
package main.commands.slashcommands.commands.dev;

import main.commands.slashcommands.SlashCommandManager;
import main.main.Robertify;
import main.utils.component.interactions.AbstractSlashCommand;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import netscape.javascript.JSException;
import org.apache.http.HttpException;
import org.apache.http.HttpStatus;
import org.jetbrains.annotations.NotNull;
import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.IOException;

public class PostCommandInfoCommand extends AbstractSlashCommand {
    final Logger logger = LoggerFactory.getLogger(PostCommandInfoCommand.class);

    @Override
    protected void buildCommand() {
        setCommand(
                getBuilder()
                        .setName("postcommandinfo")
                        .setDescription("Post all command info to the Robertify API!")
                        .setDevCommand()
                        .build()
        );
    }

    @Override
    public String getHelp() {
        return null;
    }

    @Override
    public void onSlashCommandInteraction(@NotNull SlashCommandInteractionEvent event) {
        if (!devCheck(event)) return;

        SlashCommandManager slashCommandManager = new SlashCommandManager();
        final var commands = slashCommandManager.getCommands();

        final JSONObject body = new JSONObject();
        final JSONArray cmdsArr = new JSONArray();

        int id = 0;
        for (final var cmd : commands) {
            cmdsArr.put(new JSONObject()
                    .put("id", id++)
                    .put("name", cmd.getName())
                    .put("description", cmd.getDescription())
                    .put("category", slashCommandManager.getCommandType(cmd).name().toLowerCase())
            );
        }

        body.put("commands", cmdsArr);

        event.deferReply().queue();
        try (final var response = Robertify.getRobertifyAPI().postCommandInfo(body)) {
            if (response.code() == HttpStatus.SC_CREATED)
                event.getHook().sendMessage("Posted!")
                        .setEphemeral(true)
                        .queue();
            else {
                try {
                    final var errMsg = new JSONObject(response.body().string()).getString("message");

                    event.getHook().sendMessage("There was an issue attempting to post commands! Check console for more information.\n\nError Message: `"+errMsg+"`\nError Code: `"+response.code()+"`")
                            .setEphemeral(true)
                            .queue();
                    logger.error("There was an error when trying to POST commands: " + errMsg);
                } catch (JSONException e) {
                    event.getHook().sendMessage("There was an issue attempting to post commands! Check console for more information.\n\nError Code: `"+response.code()+"`")
                            .setEphemeral(true)
                            .queue();
                    logger.error(response.body().string());
                }

            }
        } catch (Exception e) {
            logger.error("Unexpected error", e);
        }
    }
}

File: src/main/kotlin/main/commands/slashcommands/dev/PostCommandInfoCommand.kt
package main.commands.slashcommands.dev

import dev.minn.jda.ktx.util.SLF4J
import io.ktor.client.statement.*
import io.ktor.http.*
import main.commands.slashcommands.SlashCommandManagerKt
import main.main.RobertifyKt
import main.utils.RobertifyEmbedUtilsKt.Companion.sendEmbed
import main.utils.component.interactions.slashcommand.AbstractSlashCommandKt
import main.utils.component.interactions.slashcommand.models.CommandKt
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import org.json.JSONArray
import org.json.JSONException
import org.json.JSONObject

class PostCommandInfoCommand : AbstractSlashCommandKt(
    CommandKt(
        name = "postcommandinfo",
        description = "Post all command info to the Robertify API!",
        developerOnly = true
    )
) {

    companion object {
        private val logger by SLF4J
    }

    override suspend fun handle(event: SlashCommandInteractionEvent) {
        val commandManager = SlashCommandManagerKt
        val commands = commandManager.globalCommands

        val body = JSONObject()
        val cmdArr = JSONArray()

        commands.forEachIndexed { i, command ->
            cmdArr.put(
                JSONObject()
                    .put("id", i)
                    .put("name", command.info.name)
                    .put("description", command.info.description)
                    .put("category", commandManager.getCommandType(command)!!.name.lowercase())
            )
        }

        body.put("commands", cmdArr)
        event.deferReply(true).queue()

        val guild = event.guild
        val response = RobertifyKt.externalApi.postCommandInfo(body)
        if (response != null && response.status == HttpStatusCode.Created)
            event.hook.sendEmbed(guild, "Posted!")
                .queue()
        else if (response != null) {
            try {
                val err = JSONObject(response.bodyAsText()).getString("message")
                event.hook.sendEmbed(
                    guild,
                    "There was an issue attempting to post commands! Check console for more information.\n\nError Message:\n```$err```\nError Code:`${response.status.value}`"
                )
                    .queue()
                logger.error("There was an error when trying to POST commands: $err")
            } catch (e: JSONException) {
                event.hook.sendEmbed(
                    guild,
                    "There was an issue attempting to post commands! Check console for more information.\n\nError Code:`${response.status.value}`"
                )
                    .queue()
                logger.error(response.bodyAsText())

            }
        } else event.hook.sendEmbed(guild, "Could not get a response! Are you sure there's a connection to the API?")
            .queue()
    }
}